lib/internal/fs/utils.js

```diff
@@ -334,6 +336,11 @@ function getOptions(options, defaultOptions = kEmptyObject) {
   if (options.signal !== undefined) {
     validateAbortSignal(options.signal, 'options.signal');
   }
+
+  if (options.recursive && typeof options.recursive !== 'boolean') {
```

Comment:
I'm not sure that this should be part of `getOptions()`. There are other fs APIs that support recursive mode, and none of those APIs have added it to this function.

Reply:
Okay I'll take a closer look